### PR TITLE
Restructure the "Evaluating models" documentation page and rename this page to "Testing models"

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- Make information about conversation tests the first thing on this page
- Make sure this information is correct (especially after all the variable names changes)
- Rename the page to “Testing models”
- Add a link to a dedicated forum category (because we expect people to ask more questions about testing)

**Status (please check what you already did)**:
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)

### CODE REVIEW

File: changelog/2237.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Restructure the "Evaluating models" documentation page and rename this page to "Testing models".
```
Comment (wochinge) [inline changelog/2237.doc.rst]: Let's link to the new page.

File: docs/user-guide/testing-models.rst [hunk 2]
```diff
@@ -38,22 +38,25 @@ Here is an example:
   * inform: in [Paris](location)
      - utter_ask_price
 
-By default Rasa saves tests to ``tests/conversation_tests.md``. You can evaluate your model
+By default Rasa saves tests to ``tests/conversation_tests.md``. You can test your model
```
Comment (wochinge) [inline docs/user-guide/testing-models.rst]:
```suggestion
By default Rasa Open Source saves tests to ``tests/conversation_tests.md``. You can test your model
```

File: docs/user-guide/testing-models.rst [hunk 3]
```diff
@@ -16,12 +16,12 @@ Evaluating Models
    If you are looking to tune the hyperparameters of your NLU model,
    check out this `tutorial <https://blog.rasa.com/rasa-nlu-in-depth-part-3-hyperparameters/>`_.
 
-.. _end_to_end_evaluation:
+.. _end_to_end_testing:
 
-End-to-End Evaluation
+End-to-End Testing
 ---------------------
```
Comment (wochinge) [inline docs/user-guide/testing-models.rst]:
```suggestion
End-to-End Testing
------------------
```

**Comment by akelad** [comment]
@alwx what is the reason behind the strict use of testing over evaluation btw? "Evaluating a model" is super common terminology. cc @TyDunn

**Comment by akelad** [comment]
also i still want to review, so please don't merge before that :D

**Comment by akelad** [comment]
@alwx also which issue does this cover? Is this one still relevant to do then? https://github.com/RasaHQ/rasa-x/issues/2239

**Comment by alwx** [comment]
@akelad it's this one: https://github.com/RasaHQ/rasa-x/issues/2237

**Comment by TyDunn** [comment]
@akelad @alwx I think the general idea is that we wanted to shift the high level language to "test your assistant", but I was in favor of keep the word "evaluate" around when it made sense (e.g. "model evaluation"). I expressed this opinion in this issue as well: https://github.com/RasaHQ/rasa-x/issues/2239

**Comment by akelad** [comment]
ok i will request some changes then :D

**Comment by wochinge** [comment]
I guess this one should potentially also go in: https://github.com/RasaHQ/rasa/pull/5467/files

File: docs/user-guide/command-line-interface.rst [hunk 4]
```diff
@@ -198,18 +198,18 @@ Additional arguments are:
 .. program-output:: rasa visualize --help
 
 
-Evaluate a Model on Test Data
+Testing a Model on Test Data
```
Comment (akelad) [inline docs/user-guide/command-line-interface.rst]: Should go back to the original - "testing on test data" sounds weird

File: docs/user-guide/command-line-interface.rst [hunk 5]
```diff
@@ -198,18 +198,18 @@ Additional arguments are:
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To evaluate your model on test data, run:
+To test your model on test data, run:
```
Comment (akelad) [inline docs/user-guide/command-line-interface.rst]: same here

File: docs/user-guide/testing-models.rst [hunk 6]
```diff
@@ -2,9 +2,9 @@
        library Rasa Core to improve the dialogue management of your contextual
        AI Assistant.
 
-.. _evaluating-models:
+.. _testing-models:
 
-Evaluating Models
+Testing Models
```
Comment (akelad) [inline docs/user-guide/testing-models.rst]: would also leave as "evaluating models"

File: docs/user-guide/testing-models.rst [hunk 7]
```diff
@@ -16,45 +16,107 @@ Evaluating Models
...
   Make sure your model file in ``models`` is a combined ``core``
   and ``nlu`` model. If it does not contain an NLU model, Core will use
   the default ``RegexInterpreter``.
 
-.. _nlu-evaluation:
+.. _nlu-testing:
 
-Evaluating an NLU Model
------------------------
+Testing an NLU Model
```
Comment (akelad) [inline docs/user-guide/testing-models.rst]: would leave this as "Evaluating an NLU Model" and "nlu-evaluation"

File: docs/user-guide/testing-models.rst [hunk 8]
```diff
@@ -188,9 +250,9 @@ near [Alexanderplatz](loc) [tonight](time)          O   loc time (2)          O
 ==================================================  ========================  ===========================
 
 
-.. _core-evaluation:
+.. _core-testing:
 
-Evaluating a Core Model
+Testing a Core Model
```
Comment (akelad) [inline docs/user-guide/testing-models.rst]: @TyDunn not too sure about this one - thoughts on whether it should be testing or evaluating? Evaluating sounds way more natural to me tbh

File: rasa/__main__.py [hunk 9]
```diff
@@ -33,7 +33,7 @@ def create_argument_parser() -> argparse.ArgumentParser:
         description="Rasa command line interface. Rasa allows you to build "
         "your own conversational assistants 🤖. The 'rasa' command "
         "allows you to easily run most common commands like "
-        "creating a new bot, training or evaluating models.",
+        "creating a new bot, training or testing models.",
```
Comment (akelad) [inline rasa/__main__.py]: would still say evaluating here

File: docs/user-guide/testing-models.rst (hunk 6 above)
Comment (alwx) [reply docs/user-guide/testing-models.rst]: I guess that was one of the main ideas behind this change.

Comment (akelad) [reply docs/user-guide/testing-models.rst]: but surely not *everything*? calling them `end-to-end tests` totally makes sense, but i don't think that means we need to say "evaluating" literally everywhere

File: docs/user-guide/testing-your-assistant.rst [hunk 10]
```diff
@@ -16,34 +15,96 @@ Evaluating Models
...
+.. _end_to_end_testing:
 
-End-to-End Evaluation
----------------------
+End-to-End Testing
+------------------
 
-Rasa Open Source lets you evaluate dialogues end-to-end, running through
-test conversations and making sure that both NLU and Core make correct predictions.
+Rasa Open Source lets you test dialogues end-to-end, by running through
```
Comment (TyDunn) [inline docs/user-guide/testing-your-assistant.rst]:
```suggestion
Rasa Open Source lets you test dialogues end-to-end by running through
```

File: docs/user-guide/testing-your-assistant.rst [hunk 11]
```diff
@@ -16,34 +15,96 @@ Evaluating Models
...
+conversation tests and making sure that both NLU and Core make correct predictions.
```
Comment (TyDunn) [inline docs/user-guide/testing-your-assistant.rst]:
```suggestion
test conversations and making sure that both NLU and Core make correct predictions.
```

File: docs/user-guide/testing-your-assistant.rst [hunk 12]
```diff
@@ -16,34 +15,96 @@ Evaluating Models
...
+  $ rasa test
 
-By default Rasa saves tests to ``tests/conversation_tests.md``. You can evaluate your model
-against them by running:
+.. note::
 
-.. code-block:: bash
+  :ref:`custom-actions` are not executed as part of end-to-end tests. If your custom
+  actions append any events to the tracker, this has to be reflected in your end-to-end
+  tests, e.g. by adding ``slot`` events to your end-to-end story.
```
Comment (TyDunn) [inline docs/user-guide/testing-your-assistant.rst]:
```suggestion
  tests (e.g. by adding ``slot`` events to your end-to-end story).
```

**Pull request merged**